### PR TITLE
Improve marshalling performance

### PR DESCRIPTION
**Pull request by jtjeferreira**
References #1349

Before
```
[info] GrpcMarshallingBenchmark.marshall        thrpt    3  126666.347 ± 15574.678  ops/s
[info] GrpcMarshallingBenchmark.marshallStream  thrpt    3  114767.059 ±  2759.197  ops/s
```

after `concatCheap`

```
[info] GrpcMarshallingBenchmark.marshall        thrpt    3  256018.599 ± 60197.380  ops/s
[info] GrpcMarshallingBenchmark.marshallStream  thrpt    3  221998.871 ± 83522.412  ops/s
```

As @jrudolph said maybe these microoptimizations are not worth it, so feel free to close the PR

### CODE REVIEW

File: benchmarks/src/main/scala/akka/grpc/HandlerProcessingBenchmark.scala [hunk 1]
```diff
@@ -30,7 +30,7 @@ class HandlerProcessingBenchmark extends CommonBenchmark {
   val in = Source.repeat(ServerReflectionRequest()).take(10000)
   val request: HttpRequest = {
     implicit val serializer = ServerReflection.Serializers.ServerReflectionRequestSerializer
-    GrpcRequestHelpers(Uri("https://unused.example/" + ServerReflection.name + "/ServerReflectionInfo"), in)
+    GrpcRequestHelpers(Uri("https://unused.example/" + ServerReflection.name + "/ServerReflectionInfo"), Nil, in)
```
Comment (jtjeferreira) [inline benchmarks/src/main/scala/akka/grpc/HandlerProcessingBenchmark.scala:33]: looks like the `benchmarks` project is not being compiled by travis/github actions

**Comment by raboof** [comment]
OK TO TEST

File: benchmarks/src/main/scala/akka/grpc/HandlerProcessingBenchmark.scala (hunk 1 above)
Comment (raboof) [reply benchmarks/src/main/scala/akka/grpc/HandlerProcessingBenchmark.scala:33]: oops! seems like the build the `benchmark-java` subproject but not the `benchmarks`, indeed

**Review by raboof — APPROVED** [review]
> maybe these microoptimizations are not worth it, so feel free to close the PR

In general we should be a bit careful not to get too far 'in the woods' on small optimizations as it might detract from more 'big-picture' opportunities, but in this case it seems fairly targeted and not impact maintainability too much, so I'm OK with having this one in.

**Pull request merged**